synchronizer/worker/segment/errorUtil.go

```diff
@@ -24,7 +24,7 @@ func (e *SegmentError) Error() string {
 	return errorsToPrint
 }
 
-func (e *SegmentError) addError(name string, err error) {
+func (e *SegmentError) AddError(name string, err error) {
```

Comment:
if this is going to be used for largesegment, now this should be in some general util , not inside segments since large segments will also use it